src/bin/metadata.rs

```diff
@@ -0,0 +1,72 @@
+use cargo::ops::{output_metadata, OutputTo, OutputFormat, OutputMetadataOptions};
+use cargo::util::{CliResult, CliError, Config};
+use cargo::util::important_paths::find_root_manifest_for_cwd;
+
+#[derive(RustcDecodable)]
+struct Options {
+    flag_features: Vec<String>,
+    flag_manifest_path: Option<String>,
+    flag_no_default_features: bool,
+    flag_output_format: OutputFormat,
+    flag_output_path: OutputTo,
+    flag_target: Option<String>,
+    flag_verbose: bool,
+}
+
+pub const USAGE: &'static str = r#"
+Output the resolved dependencies of a project, the concrete used versions
+including overrides, in machine-readable format.
+
+Warning! This command is experimental and output format it subject to change in future.
```

Comment:
Unfortunately I don't think that this will have much effect. We currently have no method of "nightly features" in Cargo and once this lands it's essentially defacto-stable, so it just means we'll need an extra level of scrutiny for this!